tests/test_scalars.py

```diff
@@ -0,0 +1,235 @@
+    )
+    check(
+        assert_type(
+            pd.Period(
+                freq="s", year=2012, month=1, day=1, hour=12, minute=30, second=45
+            ),
+            pd.Period,
+        ),
+        pd.Period,
+    )
+    check(assert_type(pd.Period(freq="Q", year=2012, quarter=2), pd.Period), pd.Period)
+
+
+def test_period_properties() -> None:
+    p = pd.Period("2012-1-1", freq="D")
+
+    check(assert_type(p.day, int), int)
+    check(assert_type(p.day_of_week, int), int)
+    check(assert_type(p.day_of_year, int), int)
+    check(assert_type(p.dayofweek, int), int)
+    check(assert_type(p.dayofyear, int), int)
+    check(assert_type(p.days_in_month, int), int)
+    check(assert_type(p.daysinmonth, int), int)
+    check(assert_type(p.end_time, pd.Timestamp), pd.Timestamp)
+    check(assert_type(p.freqstr, str), str)
+    check(assert_type(p.hour, int), int)
+    check(assert_type(p.is_leap_year, bool), bool)
+    check(assert_type(p.minute, int), int)
+    check(assert_type(p.month, int), int)
+    check(assert_type(p.quarter, int), int)
+    check(assert_type(p.qyear, int), int)
+    check(assert_type(p.second, int), int)
+    check(assert_type(p.start_time, pd.Timestamp), pd.Timestamp)
+    check(assert_type(p.week, int), int)
+    check(assert_type(p.weekday, int), int)
+    check(assert_type(p.weekofyear, int), int)
+    check(assert_type(p.year, int), int)
+    check(assert_type(p.freq, BaseOffset), Day)
+    check(assert_type(p.ordinal, int), int)
+
+    p2 = pd.Period("2012-1-1", freq="2D")
+    check(assert_type(p2.freq, BaseOffset), Day)
+
+
+def test_periof_add_subtract() -> None:
+    p = pd.Period("2012-1-1", freq="D")
+
+    as_pd_td = pd.Timedelta(1, "D")
+    as_dt_td = dt.timedelta(days=1)
+    as_np_td = np.timedelta64(1, "D")
+    as_np_i64 = np.int64(1)
+    as_int = int(1)
+    as_period_index = pd.period_range("2012-1-1", periods=10, freq="D")
+    as_period = pd.Period("2012-1-1", freq="D")
+    scale = 24 * 60 * 60 * 10**9
+    as_td_series = pd.Series(pd.timedelta_range(scale, scale, freq="D"))
+    check(assert_type(as_td_series, TimedeltaSeries), pd.Series)
+    as_period_series = pd.Series(as_period_index)
+    as_timedelta_idx = pd.timedelta_range(scale, scale, freq="D")
+    as_nat = pd.NaT
+
+    check(assert_type(p + as_pd_td, pd.Period), pd.Period)
+    check(assert_type(p + as_dt_td, pd.Period), pd.Period)
+    check(assert_type(p + as_np_td, pd.Period), pd.Period)
+    check(assert_type(p + as_np_i64, pd.Period), pd.Period)
+    check(assert_type(p + as_int, pd.Period), pd.Period)
+    check(assert_type(p + p.freq, pd.Period), pd.Period)
+    check(assert_type(p + (p - as_period_index), pd.PeriodIndex), pd.PeriodIndex)
+    check(assert_type(p + as_td_series, PeriodSeries), pd.Series, pd.Period)
+    check(assert_type(p + as_timedelta_idx, pd.PeriodIndex), pd.PeriodIndex)
+    check(assert_type(p + as_nat, NaTType), NaTType)
+    das8 = cast(TimedeltaSeries, (as_period_series - as_period_series))
```

Comment:
get rid of the `cast`

Reply:
Yes, the result is a `Series` of offsets.  But anything we write in a test function should be code that a user could write, and they wouldn't have access to `TimedeltaSeries`.  And it's not the right cast anyway.

My suggestion is as follows:
1. Create a new type in `core/series.pyi` that is `OffsetSeries`
2. For any of the subtraction operators in here that return a series of offsets, have them return `OffsetSeries` instead.
3. Then  you can modify `__add__()` in here so that if you are adding a `Period` and an `OffsetSeries`, you get a `PeriodSeries`
4. You could also modify `OffsetSeries` to handle addition/subtraction of `Period` to `OffsetSeries`

Reply:
I tried to do this but it broke things elsewhere in `Series`.  I think this is too complex a route, at least for this PR.  I've left the changes in but will likely just disable the test.

Reply:
Found this issue and now working.